### DISCUSSION TITLE
Segmentation Network for Sparse data, pos=pos, x=None, y=len(n_points)

### DISCUSSION THREAD

**Discussion by davodogster** [pyg-team/pytorch_geometric#4434] [Q&A]
I'm struggling to find a network for my task even though my task is a common one.

I just want to do point cloud semantic segmentation. I don't have normals, and each point has a class label.
Preferably the network can generate graphs on the fly in the forward pass using:
        edge_index = knn_graph(pos, k=K, batch=batch, loop=LOOP) 

Is it true that I have to use messagePassing if each example has a different number of samples (points)? Sparse data - but the mean number of points per example / point cloud = 50,000.
It would be great to use pointTransformer or similar, but it doesn't use MessagePassing.

Is there any existing code/network I can use in the pyG documentation?
Thank you

**Comment by rusty1s** [comment]
Thanks for your interest. We have a few examples regarding segmentation of point clouds, e.g., [here](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/pointnet2_segmentation.py) and [here](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/point_transformer_segmentation.py). Note that normals are definitely not required for these examples. `PointConv` can make use of `x=None` inside message passing. `PointTransformerConv` would require that you need to create dummy input features first, e.g., via `torch.ones(data.num_nodes, 1)`.

All these examples work for varying number of points and make use of `MessagePassing` under the hood.

**Reply by davodogster** [reply]
Thanks @rusty1s , I think those two links are for the same script. 

1. It appears Point _Transformer_Segmentation.py has builtin code to automatically create the dummy features:
![image](https://user-images.githubusercontent.com/46079516/162536039-a0ac9f4e-5391-4e17-aa30-be729b12815e.png)

2. When I feed my segmentation data to the model it throws this error:
![image](https://user-images.githubusercontent.com/46079516/162541874-6e406691-d7f6-4aed-916d-76672d3e4e48.png)

![image](https://user-images.githubusercontent.com/46079516/162536580-d3fe5c81-8028-4283-b388-475a19ae7d70.png)
.
.
.
![image](https://user-images.githubusercontent.com/46079516/162536513-e0f09495-bf03-4bf1-958e-71b2c2ab642a.png)

My data:
![image](https://user-images.githubusercontent.com/46079516/162539275-b39d935e-4e1a-4bad-9a1c-149e4f5be8a8.png)

Any advice would be greatly appreciated thank you

**Reply by rusty1s** [reply]
Fixed the link above. You need to initialize your model with `in_channels=1`, that is `Net(1, …)`.
